controllers/oscmachine_keypair_controller.go

```diff
@@ -0,0 +1,116 @@
+package controllers
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+
+	osc "github.com/outscale/osc-sdk-go/v2"
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/services/security"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func checkKeypairFormatParameters(machineScope *scope.MachineScope) (string, error) {
+	machineScope.Info("Check Keypair parameters")
+
+	var keypairSpec *infrastructurev1beta1.OscKeypair
+	nodeSpec := machineScope.GetNode()
+	if nodeSpec.KeyPair.Name == "" {
+		nodeSpec.SetKeyPairDefaultValue()
+		keypairSpec = &nodeSpec.KeyPair
+	} else {
+		keypairSpec = machineScope.GetKeypair()
+	}
+
+	keypairName := keypairSpec.Name + "-" + machineScope.GetUID()
+	keypairTagName, err := tag.ValidateTagNameValue(keypairName)
+	if err != nil {
+		return keypairTagName, err
+	}
+
+	publicKey := keypairSpec.PublicKey
+	if publicKey == "" {
+		return keypairTagName, fmt.Errorf("keypair Public Ip is empty")
+	} else {
+		if _, err := b64.StdEncoding.DecodeString(publicKey); err != nil {
+			return keypairTagName, fmt.Errorf("Invalid keypairType")
+		}
```

Comment:
Please check that decode string is a publicKey ?
And not just some encoding string ?